src/rapids_dependency_file_generator/schema.json

```diff
@@ -156,6 +158,28 @@
             },
             "additionalProperties": false,
             "required": ["pip"]
+        },
+        "extras": {
+            "type": "object",
+            "properties": {
+                "table": {
+                    "type": "string",
+                    "enum": ["build-system", "project", "project.optional-dependencies"]
+                },
+                "key": {"type": "string"}
+            },
+            "if": {
+                "properties": { "table": { "const": "project.optional-dependencies" } }
+            },
+            "then": {
+                "required": ["key"]
+            },
+            "else": {
+                "not": {
+                    "required": ["key"]
+                }
+            },
```

Comment:
Properties are not required by default. Is the "else"-block really necessary?

Reply:
From some Googling, I discovered that this not-required idiom is not what it sounds like colloquially. It doesn't mean that "key" is not required, it actually _forbids_ the key from existing. I think that is what we want since otherwise dependencies.yaml could produce invalid pyproject.toml files by providing the wrong key for a particular section (e.g. putting "dependencies" in the "build-requires" table instead of "requires").

Reply:
Ok, might be good to add an _invalid_ example to test that.

Reply:
Added a test.